### PR TITLE
@types/jest: added setTimeout method

### PR DESCRIPTION
**Pull request by totano**
Please fill in this template.

- [X] Use a meaningful title for the pull request. Include the name of the package modified.
- [X] Test the change in your own code. (Compile and run.)
- [X] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [ ] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [X] Provide a URL to documentation or source code which provides context for the suggested changes: http://facebook.github.io/jest/docs/en/jest-object.html#jestsettimeouttimeout

### CODE REVIEW

File: types/jest/index.d.ts [hunk 1]
```diff
@@ -153,6 +153,11 @@ declare namespace jest {
      * for the specified module.
      */
     function setMock<T>(moduleName: string, moduleExports: T): typeof jest;
+     /**
+     * Set the default timeout interval for tests and before/after hooks in milliseconds.
+     * Note: The default timeout interval is 5 seconds if this method is not called.
+     */
+    function setTimeout(timeout: number): typeof jest
```
Comment (ikatyang) [inline types/jest/index.d.ts]: Can you add a test for it? e.g. `jest.setTimeout(1000);`.

And you have to fix the linter error.

File: types/jest/index.ts [hunk 2]
```diff
@@ -153,6 +153,11 @@ declare namespace jest {
      * for the specified module.
      */
     function setMock<T>(moduleName: string, moduleExports: T): typeof jest;
+     /**
+      * Set the default timeout interval for tests and before/after hooks in milliseconds.
+      * Note: The default timeout interval is 5 seconds if this method is not called.
+      */
+    function setTimeout(timeout: number): typeof jest;
```
Comment (wsmd) [inline types/jest/index.ts]: The documentation comment got indented by one extra space in your last commit. Do you mind fixing that? 😄

**Review by wsmd — CHANGES_REQUESTED** [review]
Nice work. One suggestion about indentation. I'm gonna have to ask other reviewers if they have any objections to renaming the index file.

**Comment by wsmd** [comment]
I see that the tests are failing:

```
Found 3830 typings folders in /home/travis/build/DefinitelyTyped/DefinitelyTyped/types
{ Error: ENOENT: no such file or directory, open '/home/travis/build/DefinitelyTyped/DefinitelyTyped/types/jest/index.d.ts'
  errno: -2,
  code: 'ENOENT',
  syscall: 'open',
  path: '/home/travis/build/DefinitelyTyped/DefinitelyTyped/types/jest/index.d.ts' }
npm ERR! Test failed.  See above for more details.
```

**Review by wsmd — CHANGES_REQUESTED** [review]
> Can you add a test for it? e.g. jest.setTimeout(1000);

Could I ask you to add a test? Making sure we don't miss that.

File: types/jest/index.d.ts [hunk 3]
```diff
@@ -153,9 +153,14 @@ declare namespace jest {
...
-     * Creates a mock function similar to jest.fn but also tracks calls to object[methodName]
-     */
+     /**
+      * Set the default timeout interval for tests and before/after hooks in milliseconds.
+      * Note: The default timeout interval is 5 seconds if this method is not called.
+      */
+    function setTimeout(timeout: number): typeof jest;
+     /**
+      * Creates a mock function similar to jest.fn but also tracks calls to object[methodName]
+      */
```
Comment (wsmd) [inline types/jest/index.d.ts]: @totano I meant to ask you to push the indentation one space back to keep it consistent with the rest of the file

```diff
      * for the specified module.
      */
     function setMock<T>(moduleName: string, moduleExports: T): typeof jest;
-    /**
-     * Creates a mock function similar to jest.fn but also tracks calls to object[methodName]
-     */
+    ←/**
+     ←* Set the default timeout interval for tests and before/after hooks in milliseconds.
+     ←* Note: The default timeout interval is 5 seconds if this method is not called.
+     ←*/
+    function setTimeout(timeout: number): typeof jest;
+    ←/**
+     ←* Creates a mock function similar to jest.fn but also tracks calls to object[methodName]
+     ←*/
     function spyOn<T extends {}, M extends keyof T>(object: T, method: M): SpyInstance<T[M]>;
     /**
      * Indicates that the module system should never return a mocked version of
```

**Review by wsmd — CHANGES_REQUESTED** [review]

**Comment by totano** [comment]
indentation fixed. tomorrow i will add a test

File: types/jest/jest-tests.ts [hunk 4]
```diff
@@ -239,6 +239,17 @@ describe('Assymetric matchers', () => {
     });
 });
 
+describe('setTimeout', () => {
+    it('works as expected', done => {
+        jest.setTimeout(1000);
+
+        setTimeout(() => {
+                expect(true).toBeTruthy();
```
Comment (wsmd) [inline types/jest/jest-tests.ts]: Thanks for adding the test. Can we change the indentation to 4 spaces in this block for consistency?

**Review by wsmd — CHANGES_REQUESTED** [review]

**Review by wsmd — APPROVED** [review]
Great work!

**Pull request merged**